server/server.ts

```diff
@@ -471,11 +471,17 @@ async function endGameCleanup(roomId: string, game: InternalState, winningPlayer
   }, 10000);
 }
 
-async function updateRoomConfig(game: InternalState, roomId: string) {
+async function updateRoomConfig(game: InternalState, roomId: string, playerNewNickname?: InternalPlayer) {
   const roomConfig: RoomConfig = {
     capacity: 0,
     winningScore: game.winningScore,
-    playerNicknameMap: Object.fromEntries(game.players.map((player) => [player.id, player.nickname ?? player.id])),
+    playerNicknameMap: Object.fromEntries(game.players.map((player) => {
+      if (playerNewNickname && player.id == playerNewNickname.id) {
```

Comment:
Sometimes there were extra cycles happening because new nickname wasn't coming through in `game.players`, so explicitly passing player info to avoid any extra update cycles